scripts/installer/postgres-setup

```diff
@@ -0,0 +1,510 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use Getopt::Long;
+use Socket;
+use File::Copy;
+#for timing purposes
+
+my $verbose;
+my $pg_only;
+my $force;
+
+my @CONFIG_VARIABLES;
+my $WORKING_OS = "";
+my $postgresonly = 0;
+
+    @CONFIG_VARIABLES =
+      ( 'POSTGRES_SERVER', 'POSTGRES_PORT', 'POSTGRES_DATABASE', 'POSTGRES_USER', 'POSTGRES_PASSWORD', 'POSTGRES_ADMIN_PASSWORD' );
+    
+
+my %CONFIG_DEFAULTS = (
+
+    'POSTGRES_ADMIN_PASSWORD', 'secret',
+    'POSTGRES_SERVER',   '127.0.0.1',
+    'POSTGRES_PORT',     5432,
+    'POSTGRES_DATABASE', 'dvndb',
+    'POSTGRES_USER',     'dvnapp',
+    'POSTGRES_PASSWORD', 'secret',
+);
+
+
+my %CONFIG_PROMPTS = (
+    'POSTGRES_SERVER',   'Postgres Server Address',
+    'POSTGRES_PORT',     'Postgres Server Port',
+    'POSTGRES_ADMIN_PASSWORD', 'Postgres ADMIN password',
+    'POSTGRES_DATABASE', 'Name of the Postgres Database',
+    'POSTGRES_USER',     'Name of the Postgres User',
+    'POSTGRES_PASSWORD', 'Postgres user password',
+);
+
+    $CONFIG_DEFAULTS{'POSTGRES_PORT'} = $ENV{'DB_PORT'};
+    $CONFIG_DEFAULTS{'POSTGRES_SERVER'} = $ENV{'DB_HOST'};
+    $CONFIG_DEFAULTS{'POSTGRES_DATABASE'} = $ENV{'DB_NAME'};
+    $CONFIG_DEFAULTS{'POSTGRES_USER'} = $ENV{'DB_USER'};
+    $CONFIG_DEFAULTS{'POSTGRES_PASSWORD'} = $ENV{'DB_PASS'};
+# Supported Posstgres JDBC drivers:
+# (have to be configured explicitely, so that Perl "taint" (security) mode
+# doesn't get paranoid)
+# 3. CHECK POSTGRES AND JQ AVAILABILITY:
+
+my $pg_local_connection = 0;
+my $psql_exec;
+my $jq_exec = "";
+my $pg_major_version = 0;
+my $pg_minor_version = 0;
+my $pod_name = "";
+$pod_name = $ENV{'MY_POD_NAME'};
+
+my $POSTGRES_SYS_UID;
+if ( $CONFIG_DEFAULTS{'POSTGRES_SERVER'} eq 'localhost' || $CONFIG_DEFAULTS{'POSTGRES_SERVER'} eq '127.0.0.1' ) 
+{
+    $pg_local_connection = 1;
+} 
+elsif ($pg_only) 
+{
+    print "In the --pg_only mode the script can only be run LOCALLY,\n";
+    print "i.e., on the server where PostgresQL is running, with the\n";
+    print "Postgres server address as localhost - \"127.0.0.1\".\n";
+    exit 1;
+}
+
+#If it is executing in a container, proceed easy with this all-in-one block
+
+
+### 3a. CHECK FOR USER postgres: (NO LONGER USED!)
+
+###print "\nChecking system user \"postgres\"... ";
+
+###my $POSTGRES_SYS_NAME = "postgres";
+###$POSTGRES_SYS_UID = ( getpwnam("postgres") )[2];
+
+# 3b. LOCATE THE EXECUTABLES, FOR jq AND psql:
+
+my $sys_path = $ENV{'PATH'};
+my @sys_path_dirs = split( ":", $sys_path );
+
+if ( $pod_name ne "start-glassfish")
```

Comment:
While running `vagrant up` (I'm on 19bdc99) I noticed  some Perl warnings on line 87 (and on some lines below):

```
==> standalone: Use of uninitialized value $pod_name in string ne at ./postgres-setup line 87.
==> standalone: Use of uninitialized value $pod_name in string eq at ./postgres-setup line 108.
==> standalone: Use of uninitialized value $pod_name in string eq at ./postgres-setup line 118.
==> standalone: 
==> standalone: 
==> standalone: Found Postgres psql command, version 9.6.9.
==> standalone: 
==> standalone: (We'll be Using psql version 9.6)
==> standalone: Checking if we can talk to Postgres as the admin user...
==> standalone: Use of uninitialized value $ENV{"PREPOPULATE"} in numeric eq (==) at ./postgres-setup line 243.
==> standalone: Yes, we can!
==> standalone: 
==> standalone: Configuring Postgres Database:
==> standalone: Use of uninitialized value $pod_name in string eq at ./postgres-setup line 351.
```